test/common/test_centroid.cpp

```diff
@@ -324,6 +332,14 @@ TEST (PCL, compute3DCentroidCloudIterator)
 
     EXPECT_EQ (8, compute3DCentroid (it, centroid_f));
     EXPECT_EQ_VECTORS (Eigen::Vector4f (0.f, 0.f, 0.f, 1.f), centroid_f);
+
+    centroid_f = Eigen::Vector4f::Ones ();
+    indices.clear ();
+    indices[0] = cloud.points.size () - 1;
```

Comment:
This is UB. Please use `push_back` instead

Reply:
oops.. yes, I've fixed this now